### PR TITLE
REST: Use apache/iceberg-rest-fixture docker image

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Also update the naming from `iceberg-rest-adapter-image` -> `iceberg-rest-fixtures` in the docs

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
waiting to publish image first to the dockerhub

**Comment by Fokko** [comment]
@ajantha-bhat Should we split out the renaming into a separate PR? I think it would be good to get that cleaned up ASAP. We can combine the rename with decoupling the user and the repository:

https://github.com/apache/iceberg/blob/7c7b4bac9097217f5563338dbc4b30d1559b5d76/.github/workflows/publish-docker.yml#L51-L55

We want to set the repository to `apache` instead of the user. If you're using a personal dockerhub account the username and the repository are the same, if you're using an organization, they are different.

I think as part of the naming, we should also include `iceberg-rest-fixtures` in the name of the action:

https://github.com/apache/iceberg/blob/7c7b4bac9097217f5563338dbc4b30d1559b5d76/.github/workflows/publish-docker.yml#L20

**Comment by Fokko** [comment]
Pinged INFRA on the [ticket](https://issues.apache.org/jira/browse/INFRA-26275).

**Comment by ajantha-bhat** [comment]
@Fokko: Thanks for pinged on the infra ticket, I saw that repo is created now 👍 

I will rebase/retrigger this PR once we publish our docker image.

**Comment by Fokko** [comment]
@ajantha-bhat Go for it 🚀  https://hub.docker.com/repository/docker/apache/iceberg-rest-fixture/general

**Review by Fokko — APPROVED** [review]

**Comment by ajantha-bhat** [comment]
@Fokko: The build is not passing. I figuring out :( 

Locally I can't run it (has some other error). 

That is why it is still in draft.

**Comment by Fokko** [comment]
Hmm, that's too bad. Let me check if it works on my machine

**Comment by Fokko** [comment]
@ajantha-bhat I noticed that the docker-container is very old, can you try:

```
--- a/kafka-connect/kafka-connect-runtime/docker/docker-compose.yml
+++ b/kafka-connect/kafka-connect-runtime/docker/docker-compose.yml
@@ -48,8 +48,12 @@ services:
     ports:
       - 8181:8181
     environment:
+      - AWS_ACCESS_KEY_ID=admin
+      - AWS_SECRET_ACCESS_KEY=password
       - AWS_REGION=us-east-1
       - CATALOG_WAREHOUSE=s3://bucket/warehouse/
+      - CATALOG_CATALOG__IMPL=org.apache.iceberg.jdbc.JdbcCatalog
+      - CATALOG_URI=jdbc:sqlite:file:/tmp/iceberg_rest_mode=memory
       - CATALOG_IO__IMPL=org.apache.iceberg.aws.s3.S3FileIO
       - CATALOG_S3_ENDPOINT=http://minio:9000
       - CATALOG_S3_PATH__STYLE__ACCESS=true
```

**Comment by ajantha-bhat** [comment]
@Fokko: Thanks for running locally. Kafka connect checks passed now after adding catalog URI.

File: kafka-connect/kafka-connect-runtime/docker/docker-compose.yml [hunk 1]
```diff
@@ -50,6 +50,7 @@ services:
     environment:
       - AWS_REGION=us-east-1
       - CATALOG_WAREHOUSE=s3://bucket/warehouse/
+      - CATALOG_URI=jdbc:sqlite:file:/tmp/iceberg_rest_mode=memory
```
Comment (ajantha-bhat) [inline kafka-connect/kafka-connect-runtime/docker/docker-compose.yml:53]: Maybe the testcase is written such a way that, tables needs to be persisted. 
Hence, it is passing with file based URI and failing with default URI. 

Might have to refactor testcase. But locally I am not able to run these tests on MAC. 
So, using the workaround.

**Comment by ajantha-bhat** [comment]
Build has passed after changing the default URI to file based.

**Comment by ajantha-bhat** [comment]
@Fokko: Anything else needed for this PR?

**Pull request merged**